shell/platform/embedder/embedder.cc

```diff
@@ -1249,6 +1249,72 @@ void PopulateAOTSnapshotMappingCallbacks(
   }
 }
 
+std::function<void(const FlutterSemanticsUpdate*)>
+CreateEmbedderSemanticsUpdateCallback(const FlutterProjectArgs* args,
+                                      void* user_data) {
+  if (SAFE_ACCESS(args, update_semantics_callback, nullptr) != nullptr) {
+    return [ptr = args->update_semantics_callback,
+            user_data](const FlutterSemanticsUpdate* update) {
+      ptr(update, user_data);
+    };
+  }
+
+  // Map the legacy callbacks back to the modern one.
+  FlutterUpdateSemanticsNodeCallback update_semantics_node_callback = nullptr;
+  if (SAFE_ACCESS(args, update_semantics_node_callback, nullptr) != nullptr) {
+    update_semantics_node_callback = args->update_semantics_node_callback;
+  }
+
+  FlutterUpdateSemanticsCustomActionCallback
+      update_semantics_custom_action_callback = nullptr;
+  if (SAFE_ACCESS(args, update_semantics_custom_action_callback, nullptr) !=
+      nullptr) {
+    update_semantics_custom_action_callback =
+        args->update_semantics_custom_action_callback;
+  }
+
+  if (update_semantics_node_callback == nullptr &&
+      update_semantics_custom_action_callback == nullptr) {
+    return nullptr;
+  }
+
+  return [update_semantics_node_callback,
+          update_semantics_custom_action_callback,
+          user_data](const FlutterSemanticsUpdate* update) {
```

Comment:
Below is the same logic to call the old embedder callbacks (though semantic nodes have already been converted to embedder semantic nodes): 

https://github.com/flutter/engine/blob/243019f734a29acf13c87dfbf9dc12ee03a2d54d/shell/platform/embedder/embedder.cc#L1425-L1507